crates/common/src/lib.rs

```diff
@@ -511,6 +511,10 @@ impl StarknetVersion {
         StarknetVersion(Some(format!("{major}.{minor}.{patch}")))
     }
 
+    pub fn new_from_string(version: Option<String>) -> Self {
+        Self(version)
+    }
+
```

Comment:
Could also be a `From<Option<String>>` 🤷 Not overly important, but I recall it would have made my PR for this easier because all the tests had `starknet_version: None` which is pretty descriptive in a way.

Anyway, this is probably fine.